### PR TITLE
ci: change goerli tests to sepolia

### PR DESCRIPTION
**Pull request by agostbiro**
Alchemy has deprecated Goerli: https://www.alchemy.com/blog/ethereum-goerli-testnet-deprecation

### CODE REVIEW

File: crates/edr_provider/src/data.rs [hunk 1]
```diff
@@ -3732,12 +3732,6 @@ mod tests {
             chain_id: 1,
             url: get_alchemy_url(),
         },
-        // This block has both EIP-2930 and EIP-1559 transactions
```
Comment (Wodann) [inline crates/edr_provider/src/data.rs:3735]: We should migrate this test to sepolia, as testing of a block with both an Eip-2930 and Eip-1559 transaction is what's really tested here

File: packages/hardhat-core/test/internal/hardhat-network/provider/modules/eth/methods/getTransactionByHash.ts [hunk 2]
```diff
@@ -329,61 +329,6 @@ describe("Eth module", function () {
           assert.equal(tx.from, "0x8a9d69aa686fa0f9bbdec21294f67d4d9cfb4a3e");
         });
 
-        it("should get an existing transaction from goerli", async function () {
```
Comment (Wodann) [inline packages/hardhat-core/test/internal/hardhat-network/provider/modules/eth/methods/getTransactionByHash.ts:332]: @fvictorio is this logic tested elsewhere or should we also adopt sepolia here?

**Review by Wodann — CHANGES_REQUESTED** [review]

File: crates/edr_evm/src/transaction.rs [hunk 3]
```diff
@@ -53,7 +53,8 @@ where
     }
 }
 
-/// An error that occurred while trying to construct an [`ExecutableTransaction`].
+/// An error that occurred while trying to construct an
+/// [`ExecutableTransaction`].
```
Comment (agostbiro) [inline crates/edr_evm/src/transaction.rs:57]: rustfmt choked on this for some reason, not sure why it's not a problem in main

File: crates/edr_evm/src/transaction/executable.rs [hunk 4]
```diff
@@ -31,8 +31,8 @@ pub struct ExecutableTransaction {
 }
 
 impl ExecutableTransaction {
-    /// Create an [`ExecutableTransaction`] by attempting to validate and recover
-    /// the caller address of the provided transaction.
+    /// Create an [`ExecutableTransaction`] by attempting to validate and
+    /// recover the caller address of the provided transaction.
```
Comment (agostbiro) [inline crates/edr_evm/src/transaction/executable.rs:35]: rustfmt choked on this for some reason, not sure why it's not a problem in main

File: packages/hardhat-core/test/internal/hardhat-network/provider/modules/eth/methods/getTransactionByHash.ts (hunk 2 above)
Comment (agostbiro) [reply packages/hardhat-core/test/internal/hardhat-network/provider/modules/eth/methods/getTransactionByHash.ts:332]: Done

File: crates/edr_provider/src/data.rs (hunk 1 above)
Comment (agostbiro) [reply crates/edr_provider/src/data.rs:3735]: Done. Thanks @fvictorio for sending the tx!

**Review by Wodann — APPROVED** [review]

**Pull request merged**